eng/tools/PromptConfidenceScore/Program.cs

```diff
@@ -0,0 +1,1032 @@
+            var outputFilePath = useMarkdown ? "results.md" : "results.txt";
+
+            // Add console output
+            Console.WriteLine("🔍 Running tool selection analysis...");
+            Console.WriteLine($"✅ Loaded {toolCount} tools in {executionTime.TotalSeconds:F2}s");
+            if (!string.IsNullOrEmpty(customToolsFile))
+            {
+                Console.WriteLine($"📄 Using custom tools file: {customToolsFile}");
+            }
+            else
+            {
+                Console.WriteLine("🔄 Using dynamic tool loading");
+            }
+
+            if (!string.IsNullOrEmpty(customPromptsFile))
+            {
+                Console.WriteLine($"📝 Using custom prompts file: {customPromptsFile}");
+            }
+            else
+            {
+                Console.WriteLine("📝 Using default prompts (e2eTestPrompts.md)");
+            }
+
+            // Create or overwrite the output file
+            using var writer = new StreamWriter(outputFilePath, false);
+
+            if (useMarkdown)
+            {
+                await writer.WriteLineAsync("# Tool Selection Analysis Setup");
+                await writer.WriteLineAsync();
+                await writer.WriteLineAsync($"**Setup completed:** {DateTime.Now:yyyy-MM-dd HH:mm:ss}  ");
+                await writer.WriteLineAsync($"**Tool count:** {toolCount}  ");
+                await writer.WriteLineAsync($"**Database setup time:** {executionTime.TotalSeconds:F7}s  ");
+                await writer.WriteLineAsync();
+                await writer.WriteLineAsync("---");
+                await writer.WriteLineAsync();
+            }
+
+            // Load prompts from custom file, markdown file, or JSON file as fallback
+            Dictionary<string, List<string>>? toolNameAndPrompts = null;
+
+            if (!string.IsNullOrEmpty(customPromptsFile))
+            {
+                // User specified a custom prompts file
+                if (customPromptsFile.EndsWith(".md", StringComparison.OrdinalIgnoreCase))
+                {
+                    toolNameAndPrompts = await LoadPromptsFromMarkdownAsync(customPromptsFile, isCiMode);
+                }
+                else if (customPromptsFile.EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+                {
+                    toolNameAndPrompts = await LoadPromptsFromJsonAsync(customPromptsFile, isCiMode);
+                }
+                else
+                {
+                    // Try to infer format or default to markdown
+                    toolNameAndPrompts = await LoadPromptsFromMarkdownAsync(customPromptsFile, isCiMode) ??
+                                        await LoadPromptsFromJsonAsync(customPromptsFile, isCiMode);
+                }
+            }
+            else
+            {
+                // Use default fallback logic
+                toolNameAndPrompts = await LoadPromptsFromMarkdownAsync("../../../e2eTests/e2eTestPrompts.md", isCiMode);
```

Comment:
When falling back here, effectively, this tool can only be run from a specific folder given the expected pathing?